src/CMakeLists.txt

```diff
@@ -83,121 +83,69 @@ set(COORDINATE_TYPE UniformCartesian) # TODO: Make this an option when more are
 
 configure_file(config.hpp.in generated/config.hpp @ONLY)
 
-add_library(parthenon
-  bvals/cc/bvals_cc.cpp
-  bvals/cc/bvals_cc.hpp
-  bvals/cc/flux_correction_cc.cpp
-
-  bvals/fc/bvals_fc.cpp
-  bvals/fc/bvals_fc.hpp
-  bvals/fc/flux_correction_fc.cpp
-
-  bvals/boundary_conditions.cpp
-  bvals/boundary_conditions.hpp
-
-  bvals/bvals.cpp
-  bvals/bvals.hpp
-  bvals/bvals_base.cpp
-  bvals/bvals_interfaces.hpp
-  bvals/boundary_flag.cpp
-  bvals/bvals_refine.cpp
-  bvals/bvals_var.cpp
-
-  coordinates/coordinates.hpp
-  coordinates/uniform_cartesian.hpp
-
-  driver/driver.cpp
-  driver/driver.hpp
-  driver/multistage.cpp
-  driver/multistage.hpp
-
-  interface/container_collection.cpp
-  interface/container_collection.hpp
-  interface/container_iterator.hpp
-  interface/container.cpp
-  interface/container.hpp
-  interface/metadata.cpp
-  interface/metadata.hpp
-  interface/params.hpp
-  interface/properties_interface.cpp
-  interface/properties_interface.hpp
-  interface/sparse_variable.cpp
-  interface/sparse_variable.hpp
-  interface/update.cpp
-  interface/update.hpp
-  interface/variable_pack.hpp
-  interface/variable.cpp
-  interface/variable.hpp
-
-  mesh/amr_loadbalance.cpp
-  mesh/domain.hpp
-  mesh/mesh_refinement.cpp
-  mesh/mesh_refinement.hpp
-  mesh/mesh.cpp
-  mesh/mesh.hpp
-  mesh/meshblock_tree.cpp
-  mesh/meshblock_tree.hpp
-  mesh/meshblock.cpp
-  mesh/weighted_ave.cpp
-
-  outputs/formatted_table.cpp
-  outputs/history.cpp
-  outputs/io_wrapper.cpp
-  outputs/io_wrapper.hpp
-  outputs/outputs.cpp
-  outputs/outputs.hpp
-  outputs/parthenon_hdf5.cpp
-  outputs/restart.cpp
-  outputs/vtk.cpp
-
-  parthenon/driver.hpp
-  parthenon/package.hpp
-  parthenon/parthenon.hpp
-  parthenon/prelude.hpp
-
-  pgen/default_pgen.cpp
-
-  reconstruct/characteristic.cpp
-  reconstruct/dc_inline.hpp
-  reconstruct/dc.cpp
-  reconstruct/plm_inline.hpp
-  reconstruct/plm.cpp
-  reconstruct/ppm.cpp
-  reconstruct/reconstruction.hpp
-  reconstruct/reconstruction.cpp
-
-  refinement/amr_criteria.cpp
-  refinement/amr_criteria.hpp
-  refinement/refinement.cpp
-  refinement/refinement.hpp
-
-  tasks/task_id.cpp
-  tasks/task_id.hpp
-  tasks/task_list.hpp
-  tasks/task_types.hpp
-
-  utils/buffer_utils.cpp
-  utils/buffer_utils.hpp
-  utils/change_rundir.cpp
-  utils/error_checking.hpp
-  utils/show_config.cpp
-  utils/signal_handler.cpp
-  utils/trim_string.cpp
-  utils/trim_string.hpp
-  utils/utils.hpp
-
-  argument_parser.hpp
-  basic_types.hpp
-  defs.hpp
-  globals.cpp
-  globals.hpp
-  kokkos_abstraction.hpp
-  parameter_input.cpp
-  parameter_input.hpp
-  parthenon_manager.cpp
-  parthenon_manager.hpp
-  parthenon_mpi.hpp
-)
+file( GLOB base_src *.cpp )
+file( GLOB base_hdr *.hpp )
+file( GLOB bvals_src bvals/*.cpp bvals/fc/*.cpp bvals/cc/*.cpp)
```

Comment:
I would either write this a `foreach(_dir . bvals ...)` loop over the directories or use use `file(GLOB */*.cpp */*.hpp)` instead.

Reply:
Thanks I was wondering about the syntax of something like this

Reply:
Plus, please use `file(GLOB ...CONFIGURE_DEPENDS ..)`.

Reply:
Actually `file(GLOB_RECURSIVE ...)` might be better.

Reply:
I don't understand why this was switched to globs.